init_process.go

```diff
@@ -0,0 +1,406 @@
+	s3Downloader := s3manager.NewDownloader(sess)
+
+	// download file and store
+	s3FileName := strings.SplitN(s3FilePath, "/", -1)
+	fileFromS3, err := os.Create(ConfigFileFolderPath + "/" + s3FileName[len(s3FileName)-1])
+	if err != nil {
+		log.Println(err)
+		log.Fatalln("Cannot creat s3 config file to store config info")
+	}
+	defer fileFromS3.Close()
+
+	_, err = s3Downloader.Download(fileFromS3,
+		&s3.GetObjectInput{
+			Bucket: aws.String(bucketName),
+			Key:    aws.String(s3FilePath),
+		})
+	if err != nil {
+		log.Println(err)
+		log.Fatalf("Cannot download %s from s3\n", s3FileName)
+	}
+}
+
+// create a S3 client as the global S3 client for reuse
+func createS3Client() {
+	region := "us-east-1"
+	if metadataReigon != "" {
+		region = metadataReigon
+	}
+	s3Client = s3.New(session.Must(session.NewSession(&aws.Config{
+		// if not specify region here, missingregion error will raise when get bucket location
+		Region: aws.String(region),
+	})))
+
+	exists3Client = true
+}
+
+// process S3 config files user specified.
+func processS3ConfigFiles(folderPath string) {
+	fileInfos, err := ioutil.ReadDir(folderPath)
+	if err != nil {
+		log.Println(err)
+		log.Fatalf("Unable to read config files in %s folder\n", folderPath)
+	}
+
+	for _, file := range fileInfos {
+		content_b, err := ioutil.ReadFile(folderPath + "/" + file.Name())
+		if err != nil {
+			log.Println(err)
+			log.Fatalf("Cannot open file: %s\n", folderPath+"/"+file.Name())
+		}
+		content := string(content_b)
+		if strings.Contains(content, "[PARSER]") {
+			// this is a parser config file. change command
+			updateCommand("/" + folderPath + "/" + file.Name())
+		} else {
+			// this is not a parser config file. @INCLUDE
+			writeInclude("/"+folderPath+"/"+file.Name(), MainConfigFilePath)
+		}
+	}
+}
+
+// add @INCLUDE in main config file to include all customer specified config files
+func writeInclude(configFilePath, mainConfigFilePath string) {
+	mainConfigFile, err := os.OpenFile(mainConfigFilePath, os.O_APPEND|os.O_WRONLY, 0777)
+	if err != nil {
+		log.Println(err)
+		log.Fatalf("Unable to read main config file: %s\n", mainConfigFilePath)
+	}
+	defer mainConfigFile.Close()
+
+	writeContent := "@INCLUDE " + configFilePath + "\n"
+	_, err = mainConfigFile.WriteString(writeContent)
+	if err != nil {
+		log.Println(err)
+		log.Fatalf("Cannot write %s in main config file: %s\n", writeContent[:len(writeContent)-2], mainConfigFilePath)
+	}
+}
+
+// change the cammand if needed.
+// add modified command to the init_process_entrypoint.sh
+func updateCommand(parserFilePath string) {
+	FluentBitCommand = FluentBitCommand + " -R " + parserFilePath
+	log.Println("Command is change to -> " + FluentBitCommand)
+}
+
+// create a init_process_entrypoint.sh
+// which will declare ECS Task Metadata as environment variables
+// and finally invoke Fluent Bit
+func modifyInvokerFile(filePath string) {
+	invokerFile, err := os.OpenFile(filePath, os.O_APPEND|os.O_WRONLY, 0777)
```

Comment:
I see many openfiles in the code. We can have one separate function for it

Reply:
Already modified. 
After making these changes, encapsulating duplicate code. I was able to better understand how to write modular code, thank you!